migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js

```diff
@@ -0,0 +1,171 @@
+        let inProgressProjects = [];
+        let toBeDeletedRecords = [];
+
+        for (let i = 0; i < duplicateArray.length; i++) {
+          if (duplicateArray[i].status == "submitted") {
+            completedProjects.push(duplicateArray[i]);
+          } else if (duplicateArray[i].status == "started") {
+            inCompleteProjects.push(duplicateArray[i]);
+          }else if(duplicateArray[i].status == 'inProgress'){
+            inProgressProjects.push(duplicateArray[i])
+          }
+        }
+
+        completedProjects.sort(sortByCreatedAtDescending);
+        inCompleteProjects.sort(sortByCreatedAtDescending);
+        inProgressProjects.sort(sortByCreatedAtDescending);
+
+        if (completedProjects.length > 0) {
+          let firstRecord = completedProjects[0];
+
+          for (let i = 1; i < completedProjects.length; i++) {
+            toBeDeletedRecords.push(completedProjects[i]);
+          }
+
+          for (let i = 0; i < inProgressProjects.length; i++) {
+            toBeDeletedRecords.push(inProgressProjects[i]);
+          }
+
+          for (let i = 0; i < inCompleteProjects.length; i++) {
+            toBeDeletedRecords.push(inCompleteProjects[i]);
+          }
+
+        } else if(inProgressProjects.length > 0){
+            let firstRecord = inProgressProjects[0];
+
+            for (let i = 1; i < inProgressProjects.length; i++) {
+                toBeDeletedRecords.push(inProgressProjects[i]);
+              }
+
+              for (let i = 0; i < inCompleteProjects.length; i++) {
+                toBeDeletedRecords.push(inCompleteProjects[i]);
+              }
+
+        }
+          else if (inCompleteProjects.length > 0) {
+          let firstRecord = inCompleteProjects[0];
+          for (let i = 1; i < inCompleteProjects.length; i++) {
+            toBeDeletedRecords.push(inCompleteProjects[i]);
+          }
+        }
+
+        for(let record of toBeDeletedRecords)
+            {
+                try{
+
+                    let result = await db
+                    .collection("projects")
+                    .deleteOne({
+                        _id:record._id
+                    })
+                    
+                    if(result.deletedCount == 1)
+                    {
+                        successfullyDeletedRecords.push(record);
+                    }
+                    else{
+                        failedToDeletedRecords.push(record);
+                    }
+                    
+                }catch(e){
+                    
+                    failedToDeletedRecords.push(record);
+                }
+
+            }
+
+      } catch (e) {
+        console.log(e);
+        continue;
+      }
+    }
+
+    fs.writeFileSync('successfully_deleted_duplicated_records' + generateUUId()+'.js',JSON.stringify(successfullyDeletedRecords))
+    fs.writeFileSync('failed_to_deleted_duplicated_records' + generateUUId()+'.js',JSON.stringify(failedToDeletedRecords))
```

Comment:
you are creating one file with record which will have ids you can use that for validadation

Reply:
Ok Can do that.